### DISCUSSION TITLE
What is the maximum period for the implementation of a new SDK?

### DISCUSSION THREAD

**Discussion by IS-gmitov** [OP-TED/eForms-SDK#222] [Q&A] [answered]
Hi, 

What is the maximum period from the release of a new SDK version to the implementation of the new SDK by the esenders so that the XML created by their systems to be validated successfully? Is there any regulation of that?

**Comment by rousso** [comment, marked as answer]
Hi @IS-gmitov,

You should address non technical questions to the OP Helpdesk which can provide you with an official response on business topics such as this one.

From a technical perspective however, as a rule of thumb, there is no "maximum period". 
Take an example:
- Let's say you have implemented version 1.5.0. 
- Some time later 1.5.1 is released. Normally going from 1.5.0 to 1.5.1 should be automatic (for metadata-drive apps) or at least trivial (for traditional apps).
- Some time later 1.6.0 is released but you are still in 1.5.0 or 1.5.1. That is ok for as long as 1.5.x is still accepted for submitting new notices. This could be months or even years as it mostly depends on legal and business considerations rather than technical ones. That should be enough in most cases for any traditional application. A metadata-driven app should normally need no time to adjust to the new minor version of the SDK. 
- Some time later SDK 2.0.0 is released but you are still on 1.5.0. That's ok too for as long as 1.5.x is still accepted for the submission of new notices.

The time and effort you need to adapt to new SDK versions depends on whether you are creating a metadata-driven application or a traditional application:
- If your application is metadata-driven, then switching to new patches (i.e. 1.5.1) should be automatic. Switching to new minor versions should also be automatic (or at least trivial, depending on whether or not you have any internal legacy systems that depend on mappings of specific fields which may have changed). The only kind of new SDK release that you would need some time to adapt to, would be a new major version (i.e. 2.0.0) in which case you will need to make some adjustments. For major versions we will always make sure that you have enough time to adapt from a technical perspective anyway.
- If your application is not metadata-driven then there is no way to tell how long you need to adapt to any change in the SDK. It all depends on your implementation choices.

For the moment we will have to keep making adjustments to the SDK and therefore we will be making frequent releases. But by the time your application is ready to start using eForms in production, SDK releases will not be as frequent. 

Please address you question to the OP Helpdesk to get an answer from a business perspective.

Regards,
Yannis
